File: t/VB/PDFFormsDemo/PDFForms/Dialogs/GoToPageDailog.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************

Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System

Namespace PDFFormsDemo
   Partial Public Class GoToPageDailog
      Inherits Form
      Private _pagesCount As Integer
      Public Sub New(currentPageIndex As Integer, pagesCount As Integer)
         InitializeComponent()

         _pagesCount = pagesCount
         _pageNumberTextBox.Text = (currentPageIndex + 1).ToString()

         _pagesCountLabel.Text = pagesCount.ToString()
      End Sub

      Private Sub _okButton_Click(sender As Object, e As EventArgs) Handles _okButton.Click
         Dim pageNumber__1 As Integer = 0
         Integer.TryParse(_pageNumberTextBox.Text, pageNumber__1)

         If pageNumber__1 > 0 AndAlso pageNumber__1 <= _pagesCount Then
            PageNumber = pageNumber__1
         Else
            MessageBox.Show("Please insert valid page number")
            Return
         End If

         Me.DialogResult = DialogResult.OK

         Me.Close()
      End Sub

      Private Sub _cancelButton_Click(sender As Object, e As EventArgs) Handles _cancelButton.Click
         Me.DialogResult = DialogResult.Cancel

         Me.Close()
      End Sub

      Public Property PageNumber() As Integer
         Get
            Return m_PageNumber
         End Get
         Set(value As Integer)
            m_PageNumber = value
         End Set
      End Property
      Private m_PageNumber As Integer
   End Class
End Namespace

File: t/CS/PDFFormsDemo/PDFForms/Dialogs/GoToPageDailog.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace PDFFormsDemo
{
   public partial class GoToPageDailog : Form
   {
      private int _pagesCount;
      public GoToPageDailog(int currentPageIndex, int pagesCount)
      {
         InitializeComponent();

         _pagesCount = pagesCount;
         _pageNumberTextBox.Text = (currentPageIndex + 1).ToString();

         _pagesCountLabel.Text = pagesCount.ToString();
      }

      private void _okButton_Click(object sender, EventArgs e)
      {
         int pageNumber = 0;
         int.TryParse(_pageNumberTextBox.Text, out pageNumber);

         if (pageNumber > 0 && pageNumber <= _pagesCount)
            PageNumber = pageNumber;
         else
         {
            MessageBox.Show("Please insert valid page number");
            return;
         }

         this.DialogResult = DialogResult.OK;

         this.Close();
      }

      private void _cancelButton_Click(object sender, EventArgs e)
      {
         this.DialogResult = DialogResult.Cancel;

         this.Close();
      }

      public int PageNumber { get; set; }
   }
}
